sdk/storage/azblob/blob/client.go

```diff
@@ -158,6 +158,20 @@ func (b *Client) SetTier(ctx context.Context, tier AccessTier, o *SetTierOptions
 	return resp, err
 }
 
+// SetExpiry operation sets an expiry time on an existing blob. This operation is only allowed on Hierarchical Namespace enabled accounts.
+// For more information, see https://learn.microsoft.com/en-us/rest/api/storageservices/set-blob-expiry
+func (b *Client) SetExpiry(ctx context.Context, expiryType ExpiryType, o *SetExpiryOptions) (SetExpiryResponse, error) {
+	var et generated.ExpiryOptions
+	var opts *generated.BlobClientSetExpiryOptions
+	if expiryType != nil {
+		et, opts = expiryType.format(o)
+	} else {
+		et, opts = generated.ExpiryOptionsNeverExpire, nil
```

Comment:
Do we expect customers to call this with a `nil` value though?  If one creates a blob, isn't the default that it never expires?

Reply:
Yes, the expiry type is never expire when a blob is created. But let's say that if an expiry type is set on a blob, then never expire option can be used to unset the expiry. This option has to be passed explicitly to the [SetExpiry](https://learn.microsoft.com/en-us/rest/api/storageservices/set-blob-expiry#expiryoption) API.

I can think of two options here. First having `NeverExpire` as an empty struct which will implement the `ExpiryType` interface, similar to other expiry types.
```
type ExpiryTypeNeverExpire struct {
}

// implementation
func (e ExpiryTypeNeverExpire) format(o *SetExpiryOptions) (generated.ExpiryOptions, *generated.BlobClientSetExpiryOptions) {
	return generated.ExpiryOptionsNeverExpire, nil
}

// calling this
bbClient.SetExpiry(context.Background(), blob.ExpiryTypeNeverExpire{}, nil)
```

Second option is what we have currently that is having `NeverExpire` as the default expiry option.

Reply:
I understand now.  I think your original solution of `ExpiryTypeNeverExpire` is the best (sorry for the hassle).

Reply:
Made the changes.
Please note that `nil` can still be passed as `ExpiryType` since it is an interface. In this case, I am setting `ExpiryType` to `ExpiryTypeNeverExpire` as a default option.